django_lightweight_queue/management/commands/queue_runner.py

```diff
@@ -16,14 +18,51 @@ def add_arguments(self, parser):
             help="Log to the specified file.")
         parser.add_argument('--touchfile', action='store', dest='touchfile', default=None,
             help="touch(1) the specified file after running a job.")
-        parser.add_argument('--machine', action='store', dest='machine_number', default='1',
+        parser.add_argument('--machine', action='store', dest='machine_number', default=None,
             help="Machine number, for parallelism")
-        parser.add_argument('--of', action='store', dest='machine_count', default='1',
+        parser.add_argument('--of', action='store', dest='machine_count', default=None,
             help="Total number of machines running the queues")
         parser.add_argument('--only-queue', action='store', default=None,
             help="Only run the given queue, useful for local debugging")
         parser.add_argument('--config', action='store', default=None,
             help="The path to an additional django-style config file to load")
+        parser.add_argument('--exact-configuration', action='store_true',
+            help="Run queues on this machine exactly as specified. Requires the"
+                 " use of the '--config' option in addition. It is an error to"
+                 " use this option together with either '--machine' or '--of'.")
+
+    def validate_and_normalise(self, options):
+        if options['exact_configuration']:
+            if not options['config']:
+                raise CommandError(
+                    "Must provide a value for '--config' when using "
+                    "'--exact-configuration'.",
+                )
+
+            invalid_others = ('machine_count', 'machine_number', 'only_queue')
+
+            if any(options[name] is not None for name in invalid_others):
+                raise CommandError(
+                    "Must not specify '--machine', '--of' or '--only-queue'"
+                    " when using '--exact-configuration'.",
+                )
+
+        else:
```

Comment:
I was meaning can we ensure the values for options[name] are guaranteed to be an int through argparse :). Sorry, was't super clear.